rust/src/storage/azure.rs

```diff
@@ -120,18 +156,18 @@ impl StorageBackend for ADLSGen2Backend {
     {
         debug!("Listing objects under {}", path);
         let obj = parse_uri(path)?.into_adlsgen2_object()?;
-        let account_name = self.client.account();
+        self.validate_container(&obj)?;
+
         let stream = self
-            .client
+            .container_client
             .list_blobs()
```

Comment:
@sd2k the new sdk's list_blob response builder now takes a reference to container client (see  https://github.com/Azure/azure-sdk-for-rust/blob/master/sdk/storage/src/blob/container/requests/list_blobs_builder.rs#L11), which is causing the following error:

```
rror[E0515]: cannot return value referencing temporary value
   --> rust/src/storage/azure.rs:155:9
    |
130 |           let stream = self
    |  ______________________-
131 | |             .storage_client
132 | |             .as_container_client(obj.file_system)
    | |_________________________________________________- temporary value created here
...
155 |           Ok(Box::pin(stream))
    |           ^^^^^^^^^^^^^^^^^^^^ returns a value referencing data owned by the current function
```

This is because the current Azure Storage backend is not tied to a specific container, which forces us to create new method local container client on every method call.

I don't have a good solution to address this off the top of my head, so I changed azure backend's `new` interface to take container name as argument so we can store the container client as part of the struct. This fixes the build for now, but we will get into the same problem with blob client if we are to convert `get_obj` to return a stream in the future. Let me know if you any suggestion on how to better handle this upgrade.

Reply:
@houqp Ah, that isn't fun at all. I was hoping we could solve this by returning a separate struct (e.g. `ListBlobsStream`) which wraps the existing stream and the container, and implements `Stream<Item = Result<ObjectMeta, StorageError>>`. I think this should be possible somehow, but I'm running into a similar issue as you when playing around with implementing it... do you know any `Stream` experts? :D

Reply:
ha, good call, i think returning a struct that implements `Stream` is the way to go :)